cli/lib/kontena/cli/stacks/yaml/validator_v3.rb

```diff
@@ -31,6 +31,13 @@ def initialize
         Validations::CustomValidators.load
       end
 
+      # bar -> bar
+      # foo:bar -> bar
+      # foo:bar:ro -> bar
+      def mount_point_from_volume_mapping(string)
+        string.gsub(/:(?:ro|rw)\z/, '').split(':').last
```

Comment:
See the server's [`parse_volume`](https://github.com/kontena/kontena/blob/v1.2.0/server/app/helpers/volumes_helpers.rb#L3) method for how volumes should be parsed.

There can also be other flags than just `ro` and `rw`, but they are only valid for named and bind volumes, not anonymous volumes, and thus they are always the third `:`-separated field.

* `/var/lib/foo` this form of anonymous volumes can not have flags; the mount point is the first field
* `/var/lib/foo:/var/lib/foo` this form of bind mounts supports optional flags; the mount point is the second field
* `/var/lib/foo:/var/lib/foo:ro` with optional flags
* `foo:/var/lib/foo` same thing for named volumes